pkg/connector/collaborator_cache.go

```diff
@@ -100,50 +118,55 @@ func (p *collaboratorCache) buildCache(ctx context.Context) error {
 				for _, value := range values {
 					privilegeKey := workato.PrivilegeId(keyGroup, value)
 
-					p.privilegeToUser.Set(privilegeKey, compoundUser.Id(), compoundUser)
+					err = appendCachedValue[*CompoundUser](ctx, sessionStorage, privilegeToUserCachePrefix, privilegeKey, compoundUser)
+					if err != nil {
+						return fmt.Errorf("failed to set privilege to user cache in session storage: %w", err)
+					}
 				}
 			}
 
 			// Build for folders
 			for _, folderId := range collaboratorRole.FolderIDs {
-				p.folderToUser.Set(folderId, compoundUser.Id(), compoundUser)
+				folderIdStr := strconv.Itoa(folderId)
+				err = appendCachedValue[*CompoundUser](ctx, sessionStorage, folderToUserCachePrefix, folderIdStr, compoundUser)
```

Comment:
We should use SetManyJSON and not set in a loop to reduce network requests in my opinion